src/App.js

```diff
@@ -20,29 +28,104 @@ export class App extends React.Component {
         });
     }
 
-    handleCompletedTodo = id => {
+    // rename correctly
+    completeTodo = id => {
         const todoIndex = this.state.todos.findIndex(item => item.id === id);
-        this.state.todos[todoIndex].done = !this.state.todos[todoIndex].done;
+        const todos = this.state.todos;
+        todos[todoIndex].done = !todos[todoIndex].done;
 
         this.setState({
             todos: [...this.state.todos]
         });
     }
 
-    handleDeleteTodo = id => {
-        const itemId = id;
-        const newTodos = this.state.todos.filter(item => item.id !== itemId);
+    deleteTodo = id => {
+        const newTodos = this.state.todos.filter(item => item.id !== id);
 
         this.setState({
             todos: newTodos
         });
     }
 
+    // refactor this method to have less code
+    completeAllTodos = () => {
+        const allDone = this.state.todos.every(item => item.done);
+        const allNotDone = this.state.todos.every(item => !item.done);
+        let completeTodos;
+
+        const mapAllTodos = bool => {
+            completeTodos = this.state.todos.map(item => {
+                bool ? item.done = bool : item.done = !item.done;
```

Comment:
This line of code would look better if you wrote it like this 
`item.done = bool ? bool : !item.done`